src/Otp/GoogleAuthenticator.php

```diff
@@ -72,25 +76,27 @@ public static function getKeyUri($type, $label, $secret, $counter = null, $optio
         // algorithm (currently ignored by Authenticator)
         // Defaults to SHA1
         if (array_key_exists('algorithm', $options)) {
-            $otpauth .= '&algorithm=' . $options['algorithm'];
+            $otpauth .= '&algorithm=' . rawurlencode($options['algorithm']);
         }
 
         // digits (currently ignored by Authenticator)
         // Defaults to 6
-        if (array_key_exists('digits', $options)) {
-            $otpauth .= '&digits=' . $options['digits'];
+        if (array_key_exists('digits', $options) && intval($options['digits']) !== 6 && intval($options['digits']) !== 8) {
```

Comment:
For these tiny arrays it doesn't make much of a difference, but it's a little easier to read as well as better performing if you do `isset($options['digits'])` instead of `array_key_exists('digits', $options)`
I'll merge anyways, it doesn't affect this code in functionality really.